### PR TITLE
Refs #27859 -- Added DatabaseWrapper.display_name.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: docs/releases/2.0.txt [hunk 1]
```diff
@@ -140,7 +140,9 @@ CSRF
 Database backends
 ~~~~~~~~~~~~~~~~~
 
-* ...
+* The new ``DatabaseWrapper.display_name`` attribute allows specifying the
```
Comment (timgraham) [inline docs/releases/2.0.txt]: I'd put this in "backwards incompatible changes" instead (not really incompatible but it's a change that backends should make): 
```
* Add a ``DatabaseWrapper.display_name`` attribute with the name of the
  database that your backend works with. Django may use it in various messages,
  such as in system checks.
```

**Comment by felixxm** [comment]
@timgraham Thanks for review. I think that the GIS backends should inherit `display_name` from the main `DatabaseWrapper` just like `vendor`.

File: tests/backends/tests.py [hunk 2]
```diff
@@ -55,6 +55,11 @@ def test_initialization_class_attributes(self):
             instance_attr_value = getattr(conn, instance_attr_name)
             self.assertIsInstance(instance_attr_value, class_attr_value)
 
+    @unittest.skipUnless(connection.vendor != 'unknown', 'Test only if vendor is specified')
```
Comment (timgraham) [inline tests/backends/tests.py]: The skip condition should be removed as it's not possible to run the tests with an unknown vendor, I think.

File: tests/backends/tests.py [hunk 3]
```diff
@@ -55,6 +55,11 @@ def test_initialization_class_attributes(self):
...
+    def test_initialization_display_name(self):
+        self.assertIsNotNone(connection.display_name)
```
Comment (timgraham) [inline tests/backends/tests.py]: You might assert `BaseDatabaseWrapper.display_name == 'unknown'`, that would make the `assertNotEqual` more meaningful.

Comment (timgraham) [inline tests/backends/tests.py]: I think `assertNotEqual` is sufficient -- that would be if display_name is None.

File: docs/releases/2.0.txt [hunk 4]
```diff
@@ -258,6 +258,10 @@ Database backend API
   ``datetime_extract_sql()`` methods now return only the SQL to perform the
   operation instead of SQL and a list of parameters.
 
+* Add a ``DatabaseWrapper.display_name`` attribute with the name of the
```
Comment (timgraham) [inline docs/releases/2.0.txt]: Sorry to revise my suggestion again: `Third-party database backends should add a...`

**Comment by felixxm** [comment]
@timgraham Thanks. I updated PR.

**Pull request merged**